### PR TITLE
[ros-vm-windows] add vsts agent as optional to deploy.

### PR DESCRIPTION
**Pull request by seanyen**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Add options to deploy VSTS build agent and register with a VSTS agent pool.
*
*

### CODE REVIEW

**Comment by bmoore-msft** [comment]
Check the CI results for the BP violations

**Comment by seanyen** [comment]
@bmoore-msft Thanks for the review. I cleaned up the code and now CI shows all green.

File: ros-vm-windows/InstallVstsAgent.ps1 [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+$currentLocation = Split-Path -parent $MyInvocation.MyCommand.Definition
+Write-Verbose "Current folder: $currentLocation" -verbose
+
+#Create a temporary directory where to download from VSTS the agent package (vsts-agent.zip) and then launch the configuration.
+$agentTempFolderName = Join-Path $env:temp ([System.IO.Path]::GetRandomFileName())
+New-Item -ItemType Directory -Force -Path $agentTempFolderName
+Write-Verbose "Temporary Agent download folder: $agentTempFolderName" -verbose
+
+$serverUrl = "https://dev.azure.com/$VSTSAccount"
```
Comment (bmoore-msft) [inline ros-vm-windows/InstallVstsAgent.ps1]: will this work on every instance, even if this format has not been enabled?

Comment (seanyen) [reply ros-vm-windows/InstallVstsAgent.ps1]: @bmoore-msft 

> will this work on every instance, even if this format has not been enabled?

What does that mean? Honestly, the script is shamelessly copied from https://github.com/Azure/azure-quickstart-templates/blob/master/201-vm-vsts-agent/scripts/InstallVstsAgent.ps1. I only tested it with my instances and I might not have full test coverage on everything.

Comment (bmoore-msft) [reply ros-vm-windows/InstallVstsAgent.ps1]: Azure DevOps allows accessing by

accountname.visualstudio.com
or
dev.azure.com/accountname

But I just tried an old one I have and both seem to work fine...

**Pull request merged**